File: SaasFeeGuides/SaasFeeGuides/ClientApp/src/app/viewModels/customer.ts
export interface CustomerModel {

  id: number;
  email: string;
  firstName: string;
  lastName: string;
  address: string;
  dateOfBirth: Date;
  phoneNumber: string;
}

export class Customer {

  constructor(
    public model: CustomerModel
  ) {
  }

  public formattedName() {
    return this.model.firstName + ' ' + this.model.lastName;
  }

  public formattedNameEmail() {
    return this.model.firstName + ' ' + this.model.lastName + ', ' + this.model.email;
  }
}

File: SaasFeeGuides/SaasFeeGuides/Models/Customer.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SaasFeeGuides.Models
{
    public class Customer
    {
        public int? Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string PhoneNumber { get; set; }
        public string UserId { get; set; }
        public string Address { get; set; }

    }

    public class CustomerComparer : IEqualityComparer<Customer>
    {
        public bool Equals(Customer x, Customer y)
        {
            return x.Email.Equals(y.Email, StringComparison.OrdinalIgnoreCase);
        }

        public int GetHashCode(Customer obj)
        {
            return obj.Email.GetHashCode();
        }
    }
}
